ui/src/app/views/account-creation-success/component.spec.ts

```diff
@@ -0,0 +1,89 @@
+import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {FormsModule} from '@angular/forms';
+import {RouterTestingModule} from '@angular/router/testing';
+
+import {ClarityModule} from '@clr/angular';
+
+import {
+  DataAccessLevel,
+  ProfileService,
+} from 'generated';
+
+import {ProfileServiceStub} from 'testing/stubs/profile-service-stub';
+import {ServerConfigServiceStub} from 'testing/stubs/server-config-service-stub';
+import {SignInServiceStub} from 'testing/stubs/sign-in-service-stub';
+
+import {
+  updateAndTick
+} from '../../../testing/test-helpers';
+import {ServerConfigService} from '../../services/server-config.service';
+import {SignInService} from '../../services/sign-in.service';
+
+import {AccountCreationModalsComponent} from '../account-creation-modals/component';
+import {AccountCreationSuccessComponent} from '../account-creation-success/component';
+import {AccountCreationComponent} from '../account-creation/component';
+import {LoginComponent} from '../login/component';
+import {RoutingSpinnerComponent} from '../routing-spinner/component';
+
+class AccountCreationSuccessPage {
+  fixture: ComponentFixture<AccountCreationSuccessComponent>;
+  component: AccountCreationSuccessComponent;
+
+  constructor(testBed: typeof TestBed) {
+    this.fixture = testBed.createComponent(AccountCreationSuccessComponent);
+    this.component = this.fixture.componentInstance;
+    this.readPageData();
+  }
+
+  readPageData() {
+    updateAndTick(this.fixture);
+  }
+}
+
+
+describe('AccountCreationSuccessComponent', () => {
+  let profileServiceStub: ProfileServiceStub;
+  let page: AccountCreationSuccessPage;
+  beforeEach(fakeAsync(() => {
+    profileServiceStub = new ProfileServiceStub();
+    TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule,
+        FormsModule,
+        ClarityModule.forRoot()
+      ],
+      declarations: [
+        AccountCreationModalsComponent,
+        AccountCreationSuccessComponent,
+        RoutingSpinnerComponent
+      ],
+      providers: [
+        { provide: AccountCreationComponent, useValue: {profile: {
```

Comment:
I'm not sure whether I like adding this test, just because I don't want to entrench the current approach. I'm really not convinced about this "inject the previous component" approach we're using here and would probably like this to be cleaned up. (as highlighted by the weird provider setup going on here).

Reply:
I'm also not a huge fan of this test. Its probably the test I am most hesitant about. Let me check with Ursa and Greg on it and see what they think.

Reply:
Okay! I talked to Greg, and I think it is worth having this test in order to test things where the behavior is purely scoped to the AccountCreationSuccessComponent. However, I am changing the profile value to be more of a variable/use our stub variables, rather than be hardcoded.

Reply:
To clarify, I'm fine with the test and how it is written. What I don't like is the component implementation itself and I think we should probably revisit it at some point.

Keeping the test in this PR sounds fine.